File: backend/UniversidadAPI/Models/Carrera.cs
namespace UniversidadAPI.Models
{
    public class Carrera
    {
        public int CarreraId { get; set; }
        public int FacultadId { get; set; }
        public string Nombre { get; set; } = string.Empty;
        public string? Descripcion { get; set; }
        public int DuracionSemestres { get; set; }
        public string? TituloOtorgado { get; set; }
        public DateTime FechaRegistro { get; set; }
        public bool Activo { get; set; }

        // Navigation property
        public Facultad? Facultad { get; set; }
    }
}

File: frontend/universidad/src/app/models/carrera.ts
import { Facultad } from './facultad';

export interface Carrera {
  carreraId: number;
  facultadId: number;
  nombre: string;
  descripcion?: string;
  duracionSemestres: number;
  tituloOtorgado?: string;
  fechaRegistro: Date;
  activo: boolean;
  facultad?: Facultad;
}

export interface CreateCarrera {
  facultadId: number;
  nombre: string;
  descripcion?: string;
  duracionSemestres: number;
  tituloOtorgado?: string;
}

export interface UpdateCarrera {
  nombre?: string;
  descripcion?: string;
  duracionSemestres?: number;
  tituloOtorgado?: string;
  activo?: boolean;
}
